### ISSUE TITLE
Resub crashes

### ISSUE THREAD

**Issue by Koeseb** [d0p3t/SubNotifier#7]
after some time  even when nothing happens it crashes.

`C:\Users\koeseb\Desktop\SubNotifier-master\src\lib\TwitchBot.js:202
          var resubAlertMessages = _config.Config.customMessages[channel].resubscriptions;
                                                                                                                 ^

TypeError: Cannot read property 'resubscriptions' of undefined
    at client.<anonymous> (C:/Users/koeseb/Desktop/SubNotifier-master/src/lib/TwitchBot.js:158:36)
    at client.EventEmitter.emit (C:\Users\koeseb\Desktop\SubNotifier-master\node_modules\tmi.js\lib\events.js:101:25)
    at client.EventEmitter.emits (C:\Users\koeseb\Desktop\SubNotifier-master\node_modules\tmi.js\lib\events.js:64:19)
    at client.handleMessage (C:\Users\koeseb\Desktop\SubNotifier-master\node_modules\tmi.js\lib\client.js:578:30)
    at parts.forEach (C:\Users\koeseb\Desktop\SubNotifier-master\node_modules\tmi.js\lib\client.js:959:36)
    at Array.forEach (<anonymous>)
    at client._onMessage (C:\Users\koeseb\Desktop\SubNotifier-master\node_modules\tmi.js\lib\client.js:958:11)
    at WebSocket.onMessage (C:\Users\koeseb\Desktop\SubNotifier-master\node_modules\ws\lib\WebSocket.js:418:14)
    at emitTwo (events.js:125:13)
    at WebSocket.emit (events.js:213:7)
[nodemon] app crashed - waiting for file changes before starting...`

**Comment by d0p3t** [comment]
Hi Koeseb,

Please check your `config.js` that you have at least one resubscriptions message defined for every channel you are connecting to.

For example:
```javascript
'#d0p3t': {
  subscriptions: {
    custom1: 'PogChamp SUB Thank you for subscribing {{username}}',
    custom2: 'CoolStoryBob SUB Thank you for subscribing {{username}}',
    custom3: 'DansGame SUB Thank you for subscribing {{username}}' },
  resubscriptions: {
    custom1: 'PogChamp RESUB Thank you {{username}} for resubscribing for {{months}} months! {{years}}',
    custom2: 'WutFace RESUB Thank you {{username}} for resubscribing for {{months}} months! {{years}}',
    custom3: '<3 RESUB Thank you {{username}} for resubscribing for {{months}} months! {{years}}' },
  bits: {
    custom1: 'Wowzers, {{username}} cheered {{bits}} BITS!',
    custom2: 'Oh so kind, {{username}} with the {{bits}} BITS!',
    custom3: 'LUL we got a happy cheerer over here ({{username}}: {{bits}} bits)!' },
},
```

This issue is most likely a misconfigured configuration file

**Comment by Koeseb** [comment]
I do have for both channels set a resub message

`customMessages: {
    '#trilluxe': {
      subscriptions: {
        custom1: 'trilluxeHUTAB DANKE {{username}} für deinen Sub! trilluxeHUTAB trilluxeLOGO'},
      resubscriptions: {
        custom1: 'trilluxeHUTAB DANKE {{username}} für deinen RE-Sub! trilluxeHUTAB Du bist schon seit {{years}} Jahren und {{months}} Monaten dabei! trilluxeHUTAB trilluxeLOGO'},
      bits: {
        custom1: '',
        custom2: '',
        custom3: '' },
    },
    '#sweetdreams': {
      subscriptions: {
        custom1: 'sweetdrLove Thank you for Subbing! {{username}} sweetdrLove Welcome to the DREAM TEAM sweetdrLove'},
      resubscriptions: {
        custom1: 'sweetdrLove Thank you for Re-Subbing! {{username}} sweetdrLove Welcome back to the DREAM TEAM sweetdrLove'},
      bits: {
        custom1: '',
        custom2: '',
        custom3: '' },
    },
  },`

could it be bc i have 2 configs? one in src\config and one in dist\config but both are the same

**Comment by d0p3t** [comment]
_About 2 config files: No, the `src` folder consists of the original files and `dist` has files converted for production environments (compatibility with more Javascript engines)._

Your configuration looks fine. Try to add more than one custom message for `subscriptions` and `resubscriptions`. Since you want to use 1 message, copy the first one into the second one. There might be an issues reading the messages, I will debug this tomorrow. 

Thank you for reporting this.
![bitmoji](https://render.bitstrips.com/v2/cpanel/10212734-189398152_3-s4-v1.png?transparent=1&palette=1&width=246)

**Comment by Koeseb** [comment]
All right ty!

**Comment by d0p3t** [comment]
**Update:** I've been unable to reproduce your issue. Could you please provide more information when this issue occurs?

**Comment by Koeseb** [comment]
Never happed since i changed to 3 custom msg

**Comment by d0p3t** [comment]
Thank you for your report. 

I'll keep this issue open, due to a possible bug with 1 custom message. I will fix this bug in the next update.

![bitmoji](https://render.bitstrips.com/v2/cpanel/9946208-189398152_3-s4-v1.png?transparent=1&palette=1&width=246)

**Comment by d0p3t** [comment]
Fix pushed in 1846c94

Added a check to see whether there is 1 or more custom messages in an alert type. If there is only 1, there will be no random number generated as that is unnecessary. This should prevent the alert message index chosen to ever be out of range.

Closing this issue.
